tests/BackupManager/Databases/MysqlDatabaseTest.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+namespace DigipolisGent\Tests\Robo\Task\Deploy\BackupManager\Databases;
+
+use DigipolisGent\Robo\Task\Deploy\BackupManager\Databases\MysqlDatabase;
+
+class MysqlDatabaseTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * Tests MysqlDatabase normal run.
+     */
+    public function testRun()
+    {
+        $unique = md5(uniqid());
+        $database = new MysqlDatabase();
+        $database->setConfig(
+            [
+                'type' => 'mysql',
+                'host' => 'localhost',
+                'port' => '3306',
+                'user' => 'root',
+                'pass' => 'password',
+                'database' => 'test',
+                'ignoreTables' => [],
+                'structureTables' => [],
+                'tables' => [],
+                'dataOnly' => false,
+                'orderedDump' => false,
+                'singleTransaction' => true,
+                'extra' => '--opt',
+            ]
+        );
+
+        $this->assertTrue($database->handles('mysql'));
+        $this->assertFalse($database->handles('mongodb'));
+        $this->assertFalse($database->handles($unique));
+        $this->assertEquals(
+            "(mysqldump --host='localhost' --port='3306' --user='root' "
+            . "--password='password' 'test'  --routines --single-transaction "
+            . "--opt ) > '{$unique}.sql'",
+            $database->getDumpCommandLine($unique . '.sql')
+        );
+    }
+
+    /**
+     * Tests MysqlDatabase with tables option.
+     */
+    public function testRunTables()
+    {
+        $unique = md5(uniqid());
+        $database = new MysqlDatabase();
+        $database->setConfig(
+            [
+                'type' => 'mysql',
+                'host' => 'localhost',
+                'port' => '3306',
+                'user' => 'root',
+                'pass' => 'password',
+                'database' => 'test',
+                'ignoreTables' => [],
+                'structureTables' => ['cache', 'watchdog'],
```

Comment:
Since the behaviour might change, shouldn't this for this test be empty? To purely test the tables option.